packages/provider/src/shared/validate.ts

```diff
@@ -0,0 +1,82 @@
+import type { JsonRpcResponse, JsonRpcSuccess } from 'json-rpc-engine';
```

Comment:
You can use the validation functions [here](https://github.com/MetaMask/snaps-skunkworks/blob/184b8050d685aaab3bd65f0f9923df3552c68be4/packages/utils/src/namespace.ts).